mne/io/tests/test_read_raw.py

```diff
@@ -99,3 +99,62 @@ def test_read_raw_multiple_dots(tmp_path):
     dst = tmp_path / "test.this.file.edf"
     copyfile(src, dst)
     read_raw(dst)
+
+
+@pytest.fixture(scope="module")
+def reader_excluded_from_read_raw() -> set[str]:
+    """Set of excluded readers from read_raw."""
+    return {
+        "read_raw_bti",
+        "read_raw_hitachi",
+        "read_raw_neuralynx",
+    }
+
+
+def test_all_reader_documented(reader_excluded_from_read_raw: set[str]):
+    """Test that all the readers in the documentation are accepted by read_raw."""
+    readers = _get_supported()
+    # flatten the dictionaries and retrieve the function names
+    functions = [foo.__name__ for value in readers.values() for foo in value.values()]
+    # read documentation .rst source file
+    doc_file = Path(__file__).parents[3] / "doc" / "api" / "reading_raw_data.rst"
+    assert doc_file.exists()
+    with open(doc_file) as fid:
+        doc = fid.read()
+    reader_lines = [
+        line.strip() for line in doc.split("\n") if line.strip().startswith("read_raw_")
+    ]
+    reader_lines = [
+        elt for elt in reader_lines if elt not in reader_excluded_from_read_raw
+    ]
+    missing_from_read_raw = set(reader_lines) - set(functions)
+    missing_from_doc = set(functions) - set(reader_lines)
+    if len(missing_from_doc) != 0 or len(missing_from_read_raw) != 0:
+        raise AssertionError(
+            "Functions missing from documentation:\n\t"
+            f"{'\n\t'.join(missing_from_doc)}\n\n"
+            "Functions missing from read_raw:\n\t"
+            f"{'\n\t'.join(missing_from_read_raw)}"
+        )
+    if sorted(reader_lines) != list(reader_lines):
+        raise AssertionError(
+            "Functions in documentation are not sorted. Expected order:\n\t"
+            f"{'\n\t'.join(sorted(reader_lines))}"
+        )
+
+
+def test_all_reader_documented_in_docstring():
+    """Test that all the readers are documented in read_raw docstring."""
+    readers = _get_supported()
+    # flatten the dictionaries and retrieve the function names
+    functions = [foo.__name__ for value in readers.values() for foo in value.values()]
+    doc = read_raw.__doc__.split("Parameters")[0]
+    documented = [elt.strip().split("`")[0] for elt in doc.split("mne.io.")[1:]]
+    missing_from_docstring = set(functions) - set(documented)
+    if len(missing_from_docstring) != 0:
+        raise AssertionError(
+            "Functions missing from docstring:\n\t"
+            f"{'\n\t'.join(missing_from_docstring)}"
+        )
+    if sorted(documented) != documented:
+        raise AssertionError("Functions in docstring are not sorted.")
```

Comment:
@larsoner I added those 2 tests, it did catch a couple of missing readers, 3 of which I excluded because they require a folder or an extension-less file as input.